test/domainRegistry.test.ts

```diff
@@ -0,0 +1,325 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { beforeEach } from "mocha";
+
+const ContractName = "DomainRegistry";
+
+// TEST contract ./contracts/domainRegistry.sol
+describe(ContractName, function () {
+  let domainRegistry: any;
+  let addrOwner: SignerWithAddress;
+  let addr2: SignerWithAddress;
+  let addr3: SignerWithAddress;
+  let contractAddress: SignerWithAddress;
+  const depositAmount = ethers.parseEther("0.006");
+
+  beforeEach(async function () {
+    [addrOwner, addr2, addr3] = await ethers.getSigners();
+    domainRegistry = await ethers.deployContract(ContractName);
+    contractAddress = await domainRegistry.getAddress();
+  });
+
+  it("Should top level domain special characters", async function () {
+    const invalidDomainNames = [
+      ".com",
+      "com.",
+      "-com",
+      "ua-",
+      "domain@name",
+      "special*char",
+    ];
+    for (const domainName of invalidDomainNames) {
+      await expect(
+        domainRegistry
+          .connect(addrOwner)
+          .registerPriceDomain(domainName, depositAmount)
+      ).to.be.revertedWith("It is prohibited to use special characters");
+    }
+  });
+
+  it("Should with upercase", async function () {
+    const invalidDomainNames = ["UA", "Com", "eXample"];
+
+    for (const domainName of invalidDomainNames) {
+      await domainRegistry
+        .connect(addrOwner)
+        .registerPriceDomain(domainName, depositAmount);
+      let result = await domainRegistry.getDomain(domainName);
+      expect(result.name).to.equal(domainName.toLowerCase());
+    }
+  });
+
+  it("Should register owner domain with a deposit, price and zero address", async function () {
+    const domainName = "example";
+    await domainRegistry
+      .connect(addrOwner)
+      .registerPriceDomain(domainName, depositAmount);
+    const domain = await domainRegistry.getDomain(domainName);
+    expect(domain.owner).to.equal(ethers.ZeroAddress);
+    expect(domain.deposit).to.equal(0);
+    expect(domain.price).to.equal(depositAmount);
+  });
```

Comment:
We don't have any deposit anymore according to the task. I started reading from tests and the naming confuses, alhough I don't know yet if it works in reality as a deposit (plege) or a payment.   Anyway, the naming is not the best.   -1 point